compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/FileSystemRepository.java

```diff
@@ -68,7 +68,13 @@
 
     // TODO Refactor this when we do repository/cache split
     public FileSystemRepository(Environment environment, Path cacheDirectory) {
-        this.cacheDir = cacheDirectory;
+        this.cacheDir = cacheDirectory.resolve(ProjectConstants.CACHES_DIR_NAME);
+        this.bala = cacheDirectory.resolve(ProjectConstants.REPO_BALA_DIR_NAME);
+        this.environment = environment;
+    }
+
+    public FileSystemRepository(Environment environment, Path cacheDirectory, String distributionVersion) {
+        this.cacheDir = cacheDirectory.resolve(ProjectConstants.CACHES_DIR_NAME + "-" + distributionVersion);
```

Comment:
Better to correct the classes comment with the change

Reply:
Fixed in 8a9b44d